src/simd/distances_powerpc.cc

```diff
@@ -0,0 +1,540 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#if defined(__powerpc__)
+
+#include <altivec.h>   /* Required for the Power GCC built-ins  */
+
+#include "distances_powerpc.h"
+
+#include <cmath>
+
+#define FLOAT_VEC_SIZE 4
+#define INT32_VEC_SIZE 4
+#define INT8_VEC_SIZE  16
+
+namespace faiss {
+
+float
+fvec_L2sqr_ref_ppc(const float* x, const float* y, size_t d) {
+    /* Vector implmentaion uses vector size of FLOAT_VEC_SIZE.  If the input
+       array size is not a power of FLOAT_VEC_SIZE, do the remaining elements
+       in scalar mode.  */
+    size_t i;
+    float res = 0;
+    size_t base;
+
+    vector float *vx, *vy;
+    vector float vtmp = {0, 0, 0, 0};
+    vector float vres = {0, 0, 0, 0};
+
+    base = (d / FLOAT_VEC_SIZE) * FLOAT_VEC_SIZE;
+
+    for (size_t i = 0; i < base; i = i + FLOAT_VEC_SIZE) {
```

Comment:
`for (size_t i = 0; i < base; i += FLOAT_VEC_SIZE) {`
same comment applies to loops in other functions